### PR TITLE
Add support for creating a VPC and Route 53 private hosted zone

### PR DESCRIPTION
**Pull request by hectcastro**
Make use of `majorkirby` to thread together the launching of an Amazon VPC, along with an associated Route 53 private hosted zone, with Boto.

For more details instructions, see the `README` at the root of the `deployment` subdirectory.

Connects #26. Also, tagging @kdeloach and @lliss.

### CODE REVIEW

**Comment by lliss** [comment]
Ran into an error. After running `aws --configure` and entering credentials I ran copied `default.example.yml` to `default.yml` and ran 
`python mmw_stack.py launch-stacks`  which generated the following error.

```
python mmw_stack.py launch-stacks
Traceback (most recent call last):
  File "mmw_stack.py", line 46, in <module>
    main()
  File "mmw_stack.py", line 42, in main
    mmw_config = get_config(args.mmw_config_path, args.mmw_profile)
  File "/home/likewise-open/AZVA-INT/lliss/Documents/projects/model-my-watershed/deployment/cfn/stacks.py", line 20, in get_config
    return {k: v.strip('"').strip("'") for k, v in mmw_config.items(profile)}
  File "/usr/lib/python2.7/ConfigParser.py", line 642, in items
    raise NoSectionError(section)
ConfigParser.NoSectionError: No section: 'default'
```

**Comment by lliss** [comment]
Supplied further arguments to mmw_stack.py. Made progress but the VPC fails to create.

```
python mmw_stack.py launch-stacks --aws-profile=mmw-stg --mmw-config-path=default.yml --mmw-profile=staging
[INFO] Stack VPC-34fed9de992df2f38f0276d2bd45 created
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 CREATE_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 CREATE_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 CREATE_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 CREATE_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 CREATE_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 CREATE_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_IN_PROGRESS
[DEBUG] VPC-34fed9de992df2f38f0276d2bd45 ROLLBACK_COMPLETE
[ERROR] VPC-34fed9de992df2f38f0276d2bd45 failed
```

**Comment by lliss** [comment]
BastionHostInstanceType is not being respected
Set my default.yml to use a t2.micro and it created a t2.medium

File: deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+TBD
+
+### Launch Stack
+
+After successfully creating AMIs, you can launch a Model My Watershed stack with the `launch-stacks` subcommand. To view all options for the `launch-stacks` subcommand, you can use the `--help` option.
+
+Using the parameters set in `default.yml`, you can launch a full stack with the following command:
+
+```bash
+$ ./mmw_stack.py launch-stacks --aws-profile mmw-stg --mmw-profile staging
```
Comment (kdeloach) [inline deployment/README.md]: I find this sample command useful, especially when running it for the first time. Consider adding the `--mmw-config-path` flag here.

File: deployment/mmw_stack.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    common_parser.add_argument('--aws-profile', default='default',
+                               help='AWS profile to use for launching stack '
+                                    'and other resources')
+    common_parser.add_argument('--mmw-config-path',
+                               default=os.path.join(current_file_dir,
+                                                    'default.yml'),
+                               help='Path to MMW stack config')
+    common_parser.add_argument('--mmw-profile', default='default',
+                               help='MMW stack profile to use for launching '
+                                    'stacks')
```
Comment (kdeloach) [inline deployment/mmw_stack.py]: Consider making these required arguments? If you run `./mmw_stack.py launch-stacks` without arguments you get:
```
Traceback (most recent call last):
  File "./mmw_stack.py", line 46, in <module>
    main()
  File "./mmw_stack.py", line 42, in main
    mmw_config = get_config(args.mmw_config_path, args.mmw_profile)
  File "/var/projects/model-my-watershed/deployment/cfn/stacks.py", line 20, in get_config
    return {k: v.strip('"').strip("'") for k, v in mmw_config.items(profile)}
  File "/usr/lib/python2.7/ConfigParser.py", line 642, in items
    raise NoSectionError(section)
ConfigParser.NoSectionError: No section: 'default'
```

Comment (kdeloach) [inline deployment/mmw_stack.py]: Probably not that important, since once you learn the right way to use the CLI it isn't a problem anymore.

**Comment by kdeloach** [comment]
+1 Really solid.

**Comment by hectcastro** [comment]
Issue regarding `default.yml` overrides not being respected fixed in https://github.com/WikiWatershed/model-my-watershed/commit/6a64e50a12639a3286df802cf79098625148f509.

File: deployment/README.md (hunk 1 above)
Comment (hectcastro) [inline deployment/README.md]: Addressed in https://github.com/WikiWatershed/model-my-watershed/commit/a2ba6696754f36bcdce7ae2d661c9f6822a6f5b1.

**Pull request merged**